### DISCUSSION TITLE
All branch domains redirects to Preview

### DISCUSSION THREAD

**Discussion by andresespinosapc** [vercel/community#1779] [Help]
### Summary

When I go to any branch domain it redirects to the Preview domain. I can't find any setting that could be causing this.

**Comment by amyegan** [comment]
Hi Andres. Can you share an example so I can get a better look at what is going wrong?

**Reply by andresespinosapc** [reply]
Hi! Yes, here's one of my branch urls: [cacttus-front-2rmxfbfju-cacttus-computing.vercel.app](https://cacttus-front-2rmxfbfju-cacttus-computing.vercel.app/)

**Reply by smaeda-ks** [reply]
@andresespinosapc I think it's from your middleware. Are you doing any `redirect` in your middleware?
You may want to use `req.headers.get('host')` instead of `req.nextUrl.host`.
